v3-core/contracts/libraries/TickBitmap.sol

```diff
@@ -71,8 +71,8 @@ library TickBitmap {
             initialized = masked != 0;
             // overflow/underflow is possible, but prevented externally by limiting both tickSpacing and tick
             next = initialized
-                ? (compressed + 1 + int24(BitMath.leastSignificantBit(masked) - bitPos)) * tickSpacing
-                : (compressed + 1 + int24(type(uint8).max - bitPos)) * tickSpacing;
+                ? (compressed + 1 + int24(uint24(BitMath.leastSignificantBit(masked) - bitPos))) * tickSpacing
+                : (compressed + 1 + int24(uint24(type(uint8).max - bitPos))) * tickSpacing;
```

Comment:
What is going on with these wrappers?
What happens if the expression in the function call is negative now..

Reply:
The output type of the inner expressions are uint8, they are being increased in sized to uint24, and then cast to int24. There is no information lost. Also this the way it behaves in the same way solc 0.7.6 with implicit conversion, this is important because there are cases where there are two valid ways of casting but one produces different output than the others. For example in this same case for value `255` if instead of:
casting like `uint8 -> uint24 -> int24` which produces (255 -> 255 -> 255)
I  cast as `uint8 -> int8 -> int24` that produces instead (255 -> -128 -> -128), which is not the expected value in the original Uniswap contract